fiftyone/server/routes/geo.py

```diff
@@ -28,49 +28,53 @@
 # last_modified_at is included as an argument for cache keying purposes,
 # it is not used within the function logic itself.
 # filters_key, stages_key, extended_key are JSON strings for hashability
+# For performance, we might want to batchify this so not all points are
+# stored as a single huge entry
 @alru_cache(maxsize=int(FIFTYONE_GEO_CACHE_SIZE))
 async def _fetch_geo_points(
-    dataset_name, path, filters_key, stages_key, extended_key, last_modified_at
+    dataset_name,
+    collection_name,
+    field_path,
+    filters_key,
+    stages_key,
+    extended_key,
+    last_modified_at,
 ):
+
     # deserialize keys back to Python objects for use
     filters = json.loads(filters_key) if filters_key else None
     stages = json.loads(stages_key) if stages_key else None
     extended = json.loads(extended_key) if extended_key else None
 
-    view = await fosv.get_view(
-        dataset_name,
-        stages=stages,
-        filters=filters,
-        extended_stages=extended,
-        awaitable=True,
-    )
-
-    # validate path is in schema
-    field_path = f"{path}.point"
-    if field_path not in view.get_field_schema(flat=True):
-        raise ValueError(
-            f"Path {field_path} not found in schema for dataset {dataset_name}"
+    if filters or stages or extended:
+        view = await fosv.get_view(
+            dataset_name,
+            stages=stages,
+            filters=filters,
+            extended_stages=extended,
+            awaitable=True,
         )
-
-    project_stage = {
-        "$project": {
-            "_id": 0,
-            "sampleId": {"$toString": "$_id"},
-            "coordinates": f"${field_path}.coordinates",
-        }
-    }
-
-    pipeline = view._pipeline(post_pipeline=[project_stage])
-
-    collection = foo.get_async_db_conn()[view._dataset._sample_collection_name]
-
-    agg_results = foo.aggregate(collection, pipeline)
+        pipeline = view._pipeline()
+    else:
+        pipeline = []
+
+    # only return the minimum amount of data needed to minimize network overhead
+    # if there is a regular index on the coordinates, querying this will be faster
```

Comment:
I don't think this is true. It may read from the index, but it will also read the documents
```suggestion
```